mopidy_mpd/protocol/album_art.py

```diff
@@ -0,0 +1,107 @@
+from mopidy_mpd import protocol
+from urllib.request import urlopen
+
+
+def _get_art_url(self, uri):
+    images = self.core.library.get_images([uri]).get()
+    if images[uri]:
+        largest_image = sorted(
+            images[uri], key=lambda i: i.width or 0, reverse=True
+        )[0]
+        return largest_image.uri
+
+
+cover_cache = {}
+
+
+@protocol.commands.add("albumart")
+def albumart(context, uri, offset):
+    """
+    *musicpd.org, the music database section:*
+
+        ``albumart {URI} {OFFSET}``
+
+        Locate album art for the given song and return a chunk
+        of an album art image file at offset OFFSET.
+
+        This is currently implemented by searching the directory
+        the file resides in for a file called cover.png, cover.jpg,
+        cover.tiff or cover.bmp.
+
+        Returns the file size and actual number of bytes read at
+        the requested offset, followed by the chunk requested as
+        raw bytes (see Binary Responses), then a newline and the completion code.
+
+        Example::
+
+            albumart foo/bar.ogg 0
+            size: 1024768
+            binary: 8192
+            <8192 bytes>
+            OK
+
+    .. versionadded:: 0.21
+        New in MPD protocol version 0.21
+    """
+    global cover_cache
+
+    if uri not in cover_cache:
+        art_url = _get_art_url(context, uri)
+
+        if art_url is None:
+            return b"binary: 0\n"
+
+        cover_cache[uri] = urlopen(art_url).read()
```

Comment:
A core API makes more and more sense here. This is a whole world of pain to add to Mopidy-MPD and it's really the job of each backend to provide data. Would a `get_images_data()` API be useful to other backends, I don't know (but it's not a requirement, we can do it similarly to `get_distinct`). Can mpris provide image data (https://gitlab.freedesktop.org/mpris/mpris-spec/-/issues/17)? It might also be useful in the case where the backend can't provide publicly accessible images.

Reply:
@jodal @adamcik do you have any thoughts about what to here? 
* New core api `get_image_data(uris)` with an implementation for downloading over HTTP.
* As above but as the default implementation for a matching backend extension API, allows backends to provide custom implementation I.e. Mopidy-Local would provide images directly
* above functionality part of Mopidy-MPD, anything weird or more complicated is not supported.

Reply:
One additional remark about my part of the code that was not accepted. I had foreseen a maximum (preferred) image size for images retrieved from -Local. The current code uses the largest images available (line 8), which is very inefficient (slow) for MPD clients that prefetch the images for all local albums. Ideally max preferred image size would be a config setting, in -MPD and/or -Local.

Reply:
@kingosticks I see. When I took recourse to `urlopen` in the initial implementation I indeed didn't quite consider that `mopidy-mpd` in general utilitizes calls to the core API for fetching data.

I believe your suggestion 2 sounds like the most clean way forward (new core api `get_image_data(uris)` / HTTP download per default with an option for custom implementations by backends which require so).

I am currently implementing your other feedback – thanks for the through code-review :) – I'll leave this part of the code as is for the moment, as it's likely that things will move to the core (I did however implement `ACK_ERROR_NO_EXIST` as suggested by your other above review).

Reply:
A quick note aswell: I foreseeably won't be available during the upcoming week. I'll try to upload adjustments in response to your feedback before leaving as far as possible leaving this very part out for now as described above.